### ISSUE TITLE
Null Pointer getting context

### ISSUE THREAD

**Issue by dyachera** [vaadin/spring#844]
This is related to a post I made on Stack Overflow:
https://stackoverflow.com/questions/67810653/npe-springlookupinitializer-createlookup-spring-vaadin-19

It looks like the original issue was indeed fixed, however the same issue is now occurring in SpringApplicationConfigurationFactory.  In Vaadin 14 where the related spring version works this class is not present, however in the newer version where the class is present it calls:

`ApplicationContext appContext = WebApplicationContextUtils
                .getWebApplicationContext(servletContext.getContext());`

Like the previous issue the appContext is coming back null because it doesn't do the check like implemented in SpringLookupInitializer getApplicationContext(VaadinContext context) method

**Comment by denis-anisimov** [comment]
`SpringApplicationConfigurationFactory` produces an `ApplicationConfiguration` instance.
The instance may not be asked until `ApplicationContext` is not available via `WebApplicationContextUtils.getWebApplicationContext` so there is no need to do the same thing as `SpringLookupInitializer` does since in the latter case the context may be not set _yet_ by Spring for some reasons (most likely undefined order of `ApplicationContextAware` beans execution).

So I would like to see the exact steps to reproduce which throws an NPE inside `SpringApplicationConfigurationFactory`

**Comment by pleku** [comment]
@dyachera you can e.g. fork https://github.com/vaadin/skeleton-starter-flow-spring and add the necessary code needed to reproduce the issue.

**Comment by dyachera** [comment]
I will try to create a minimal example, the project is not spring boot and rather large so could take a bit to figure out how to create a smaller version that replicates the issue. I will stick with vaadin-spring 12.4 where it works for now

**Comment by denis-anisimov** [comment]
Now I started to recall the case which required `SpringLookupInitializer` : it's not always a bean.
In non Spring boot case it will be discovered as a class by Flow `LookupServletContainerInitializer` and may be called before
Spring set the `ApplicationContext` instance to the servlet context.

And the issue which I solved inside `LookupInitializer` still may be valid: the `ApplicationContext` is set via `SpringApplicationContextInit` but it's not yet available via servlet context _still_.
And once `Lookup` is initialized by the `SpringLookupInitializer` all other initializers will be called immediately and `ApplicationContext` will not be inside servlet context.

There were no such issue in the master at all so the fix has been backported from the previous Spring add-on where `SpringApplicationConfigurationFactory` just doesn't exist.

The only thing which I don't understand: the backport should not have been fixed the problem since NPE should be thrown inside `SpringApplicationConfigurationFactory` instead of `SpringLookupInitializer` . So it should have been only part fix.

But anyway, I will correct the code to use the `ApplicationContext` instance received inside `SpringApplicationContextInit` .
